packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1116,6 +1116,14 @@ function commitLayoutEffectOnFiber(
           offscreenSubtreeIsHidden = prevOffscreenSubtreeIsHidden;
           offscreenSubtreeWasHidden = prevOffscreenSubtreeWasHidden;
         }
+
+        if (finishedWork.pendingProps.mode === 'manual') {
+          if (flags & Ref) {
+            safelyAttachRef(finishedWork, finishedWork.return);
+          }
+        } else if (finishedWork.pendingProps.mode !== undefined) {
+          safelyDetachRef(finishedWork, finishedWork.return);
```

Comment:
This also needs to be wrapped in a Ref effect. Otherwise you might forget to schedule a Ref effect in the begin phase, but it sometimes works anyway accidentally because you happen to have another effect flag scheduled during the same commit.

Reply:
Also I think this should be a regular `else` block. Not `else if`. Anything that's not manual should have its ref detached.

Reply:
> This also needs to be wrapped in a Ref effect. Otherwise you might forget to schedule a Ref effect in the begin phase, but it sometimes works anyway accidentally because you happen to have another effect flag scheduled during the same commit.

I don't follow this. What do you mean it needs to be wrapped in a Ref effect?